src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -73,9 +75,14 @@ public class RestoreCommand
         // names for central package management version information
         private const string IsCentralVersionManagementEnabled = "IsCentralVersionManagementEnabled";
 
-        public RestoreCommand(RestoreRequest request)
+        public RestoreCommand(RestoreRequest request) : this(request, new LockFileBuilderCache())
```

Comment:
I'd add the cache in the RestoreRequest, because it's additive and already where the SourceRepository is cached.

Reply:
Done